### PR TITLE
Add `humility tasks` support for RV64 s-mode

### PR DESCRIPTION
**Pull request by beezow**
Depends on https://github.com/rivosinc/hubris/pull/80

### CODE REVIEW

**Comment by beezow** [comment]
I am having some troubles with s-mode hubris, but this change appears to be working...  Probably want to wait until https://github.com/rivosinc/hubris/pull/46 is done anyways.

File: humility-core/src/arch/rv.rs [hunk 1]
```diff
@@ -85,10 +85,21 @@ impl Arch for RVArch {
...
             Ok(ptr) => Ok(core.read_word_32(ptr)? as u64),
-            // Means current task is in mscratch or sscratch, but only if on riscv
+            // Means current task is in mscratch or sscratch
             Err(_) => {
-                // TODO right now blindly check mscratch, but should check that it is valid
-                core.read_reg(Register::RiscV(RVRegister::MSCRATCH))
+                let task_register = if hubris
+                    .manifest
+                    .features
+                    .contains(&"s-mode".to_owned())
```
Comment (ghost) [inline humility-core/src/arch/rv.rs:93]: Looking at the app.toml contained in the archive for 'demo-rv64-qemu-virt-supervisor' there do not appear to be any features set for '[kernel]' (humility populates manifest.features from the kernel features). Is there another change to ensure that 's-mode' is set as a kernel feature?

Comment (ghost) [reply humility-core/src/arch/rv.rs:93]: Sorry, @tdewey-rivos has just informed me about https://github.com/rivosinc/hubris/pull/80

**Review by ghost — DISMISSED** [review]

File: humility-core/src/arch/rv.rs (hunk 1 above)
Comment (ghost) [reply humility-core/src/arch/rv.rs:93]: This is, I believe, a particularly neat solution to determining the execution mode.

**Review by ghost — DISMISSED** [review]

**Comment by beezow** [comment]
I am going to wait till https://github.com/rivosinc/humility/pull/20 is fully review and merged, then merge this in.

**Comment by beezow** [comment]
This is ready to go. I tested by running `humility tasks` while running a hubris image built from https://github.com/rivosinc/hubris/pull/80.
You can run with `humility -v tasks` if you want to verify it is reading from sscratch (beware the log output is very big for `-v`!).  You should be able to find the following output:
```
[2022-11-07T16:11:07Z TRACE humility_core::arch::rv] using sscratch
[2022-11-07T16:11:07Z TRACE humility_core::core::gdb] reading reg: RiscV(SSCRATCH)
[2022-11-07T16:11:07Z TRACE humility_core::core::gdb] checking for reg: sscratch
```

**Review by ghost — APPROVED** [review]

**Pull request merged**